### ISSUE TITLE
Can't change button text style on hover in TypeScript

### ISSUE THREAD

**Issue by ezze** [GeekyAnts/NativeBase#4428]
I can't set a style for button text on hover like described on [this docs page](https://docs.nativebase.io/pseudo-props-101):

```typescript
<Button
  _hover={{
    _text: { color: 'secondary.400' },
  }}
>
  Change My Color
</Button>
```

**To Reproduce**

Here is my code snippet:

```typescript
<Button
  variant="outline"
  borderColor="light.light-4"
  borderWidth={1}
  backgroundColor="white"
  _hover={{
    backgroundColor: 'white',
    _text: { color: 'primary.default' }
  }}
  _text={{ color: 'light.light-4' }}
  onPress={onAdd}
>
  Add
</Button>
```

I end up with TypeScript compliation error:

```
TS2322: Type '{ backgroundColor: "white"; _text: { color: string; }; }' is not assignable to type 'Omit<IPressableProps<IPressableProps<unknown>>, "_hover">'.
  Object literal may only specify known properties, and '_text' does not exist in type 'Omit<IPressableProps<IPressableProps<unknown>>, "_hover">'.
    56 |           _hover={{
    57 |             backgroundColor: 'white',
  > 58 |             _text: { color: 'primary.default' }
       |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    59 |           }}
    60 |           _text={{ color: 'light.light-4' }}
    61 |           onPress={onAdd}
```

**Expected behaviour**
My code is compiled, text style is applied on button hover.

**Platform**

- NativeBase version 3.2.2
- Device: web, Android emulator
- OS: Android SDK 29
- Browser: Google Chrome
- Browser version: Version 96.0.4664.110 (Official Build) snap (64-bit)
- React Native version: 0.64.2
- node: 14.15.1
- yarn: 1.22.17

**Comment by MD-REHMAN** [comment]
Thanks, @ezze for pointing it out. 
We'll fix it in the upcoming release.

**Comment by ankit-tailor** [comment]
On it 👀

**Comment by ezze** [comment]
@MD-REHMAN Thanks for response. Not sure if it's related but I am also struggling to apply a `_hover` state to [inputs](#4444) and [buttons](#4445).

**Comment by surajahmed** [comment]
This issue has been fixed in version `3.3.0`.

**Comment by kevinvugts** [comment]
> This issue has been fixed in version `3.3.0`.

It looks like the issue is re-introduced in version 3.3.3. Please have a look if this is really fixed. I can't get it working with the latest version. Also, the docs contains outdated version of native-base (3.2.2). Please update accordingly if possible.
